### ISSUE TITLE
jQuery Form Validator OnSuccess with Alertify not working

### ISSUE THREAD

**Issue by mittultechnobrave** [MohammadYounes/AlertifyJS#204]
I am using [jQuery Form Validator][1] with [Alertify JS][2] and I am using below code to check if my form successfully validated or not. 

    onSuccess : function($form) {
      alertify.alert('a callback will be invoked on ok.').set('onok', function(closeEvent) { 
        return true;
      }); 
      return false; // Will stop the submission of the form
    },

Here what I am trying to do is, if I press `OK` From `Alertify` then my form should be submitted. 

But as of now, if I remove ` return false; ` then my `Alertify` not showing and my form gets submited.  

I also tried to put `$('#my-form-id').submit()` inside `return true;` but yet its not working.

Basically I just want to show my `Alertify` before form gets submitted just to notify the user with my alert box. But it seems above things not working for me correctly. 

Here is the [jQuery Form Validator configuration documentation][3] which I have followed. 

Can someone guide me what I am doing wrong here ? 

  [1]: http://www.formvalidator.net/index.html
  [2]: https://alertifyjs.com/
  [3]: http://www.formvalidator.net/#configuration

**Comment by MohammadYounes** [comment]
@mittultechnobrave Alertify callbacks are asynchronous callbacks, the inner return true has no value in regards to form submission. 

 `onSuccess`  is called when the user submits the form and no input with an invalid value can be found. Thus, your code block will be executed on each valid submit.

All you need is to do is stop form submission if alert wasn't shown and allow it otherwise.
```
onSuccess : function($form) {
  if(!$form.data('alertShown')){ 
    alertify.alert('a callback will be invoked on ok.').set('onok', function(closeEvent) { 
        $form.data('alertShown', true)
        $form.submit();
    }); 
    return false; // Will stop the submission of the form
  } else {
    return true; // will allow submission, form is valid and alert was shown.
  }
}
```
